docs/services.md

```diff
@@ -266,6 +266,17 @@ In order to allow users to choose a port number for their `Services`, we must
 ensure that no two `Services` can collide.  We do that by allocating each
 `Service` its own IP address.
 
+To ensure each service receives a unique IP, an internal allocator atomically
+updates a global allocation map in etcd prior to each service. The map object
+must exist in the registry for services to get IPs, otherwise creations will
+fail with a message indicating an IP could not be allocated. A background
+controller is responsible for creating that map (to migrate from older versions
+of Kubernetes that used in memory locking) as well as checking for invalid
+assignments due to administrator intervention and cleaning up any any IPs
+that were allocated but which no service currently uses. To alter the portal
+net range, the administrator must delete the service map key and restart the
```

Comment:
----- Original Message -----
> > @@ -266,6 +266,17 @@ In order to allow users to choose a port number for
> > their `Services`, we must
> >  ensure that no two `Services` can collide.  We do that by allocating each
> >  `Service` its own IP address.
> >  
> > +To ensure each service receives a unique IP, an internal allocator
> > atomically
> > +updates a global allocation map in etcd prior to each service. The map
> > object
> > +must exist in the registry for services to get IPs, otherwise creations
> > will
> > +fail with a message indicating an IP could not be allocated. A background
> > +controller is responsible for creating that map (to migrate from older
> > versions
> > +of Kubernetes that used in memory locking) as well as checking for invalid
> > +assignments due to administrator intervention and cleaning up any any IPs
> > +that were allocated but which no service currently uses. To alter the
> > portal
> > +net range, the administrator must delete the service map key and restart
> > the
> 
> No.  If our answer to "how do you do this" is etcdctl, I do not want
> top-level docs pretending that this is a supported operation.  It's not.
> 
> Changing portal_net is either "not supported" or "only supported to expand
> the CIDR".  I think the latter should work, looking at the repair code
> here...  Please remove or repair the "To alter the portal net range"
> sentence.

The exact flow is:

1. Delete the key via etcdctl
2. Change all the masters to have the new portal_net value
3. Restart all the masters.

Is this something you think we shouldn't describe?  Or should?  To me this is
similar to "how do I migrate my keys to a new prefix", so just trying to understand
what you want me to write.